### PR TITLE
Bypass NLU pipeline when message is /intent or /intent + entities - [ENG 286]

### PR DESCRIPTION
**Pull request by varunshankar**
**Proposed changes**:
- Bypass NLU pipeline when message is /intent or /intent + entities

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Review by twerkmeister — COMMENTED** [review]
Good job so far! Can you also add the changelog file?

If we really wanted to do this right we should probably also address the following:
* This is currently difficult to test, because the `RegexMessageHandler` is still in the graph.  We should remove the RegexMessageHandler from the graph ultimately, but this also involves touching the default recipe etc, because all NLU calls currently target this component as the execution target if I remember correctly.
* We'd need to take a deeper look whether the http interpreters should actually also be covered by the regex parsing. This could be relevant if we remove the RegexMessageParser from the graph and then the http interpreters might not have any regex message parsing at all anymore. I am not really familiar with them

**Comment by varunshankar** [comment]
> Good job so far! Can you also add the changelog file?
> 
> If we really wanted to do this right we should probably also address the following:
> 
> * This change is currently difficult to test, because the `RegexMessageHandler` is still in the graph.  We should remove the RegexMessageHandler from the graph ultimately, but this also involves touching the default recipe etc, because all NLU calls currently target this component as the execution target if I remember correctly.
> * We'd need to take a deeper look whether the http interpreters should actually also be covered by the regex parsing. This could be relevant if we remove the RegexMessageParser from the graph and then the http interpreters might not have any regex message parsing at all anymore. I am not really familiar with them

Thanks a lot @twerkmeister, your review was very helpful. Is it better that I create a separate task to handle the comments that you said here?

**Comment by m-vdb** [comment]
@varunshankar given that `3.6.0` has already shipped and this is a new feature, I would suggest to merge this into `main` instead of `3.6.x`

File: tests/core/test_processor.py [hunk 1]
```diff
@@ -113,10 +114,26 @@ async def test_message_id_logging(default_processor: MessageProcessor):
...
+        # Case2: Normal user message.
+        parse_data = {
+            "text": "mocked",
+            "intent": {"name": None, "confidence": 0.0},
+            "entities": [],
+        }
+        mocked_function.return_value = parse_data
+        message = UserMessage("hi hello how are you?")
+        parsed = await default_processor.parse_message(message)
+        mocked_function.assert_called()
```
Comment (twerkmeister) [inline tests/core/test_processor.py:136]: This is really smart, I didn't know something like this was possible 😎. Very cool!

**Review by twerkmeister — APPROVED** [review]
🚀

File: tests/core/test_processor.py (hunk 1 above)
Comment (varunshankar) [reply tests/core/test_processor.py:136]: Thanks Thomas, means a lot 😄

**Pull request merged**